diff --git a/jupyter-org-client.el b/jupyter-org-client.el
index c20290f..ad0553c 100644
--- a/jupyter-org-client.el
+++ b/jupyter-org-client.el
@@ -1034,20 +1034,15 @@ parsed, wrap DATA in a minipage environment and return it."
   (with-temp-buffer
     (insert data)
     (goto-char (point-min))
-    ;; Try to determine if we are in an environment or fragment
-    (if (save-excursion
-          (forward-char 2)
-          (org-inside-LaTeX-fragment-p))
-        (org-element-latex-fragment-parser)
-      ;; If we are not in a fragment, try to parse an environment
-      (let ((env (ignore-errors
-                   (org-element-latex-environment-parser
-                    (point-max) nil))))
-        (if (eq (org-element-type env) 'latex-environment) env
-          ;; If all else fails, wrap DATA in a minipage
-          ;; environment
-          (jupyter-org-latex-environment
-           (concat "\
+    (let ((type (org-element-type (org-element-context))))
+      (cond ((eq type 'latex-fragment)
+             (org-element-latex-fragment-parser))
+            ((eq type 'latex-environment)
+             (org-element-latex-environment-parser (point-max) nil))
+            (t
+             ;; If all else fails, wrap DATA in a minipage environment
+             (jupyter-org-latex-environment
+              (concat "\
 \\begin{minipage}{\\textwidth}
 \\begin{flushright}\n" data "\n\\end{flushright}
 \\end{minipage}")))))))
